### PR TITLE
Update pam to 1.5.0.

### PR DESCRIPTION
**Pull request by ericonr**
Had to do some patching around, but the patches are fully documented.

Please test it out in whatever weird setups you have!

### CODE REVIEW

**Comment by ericonr** [comment]
Don't try this update unless you have some way of recovering, logins have stopped working for me.

**Comment by ericonr** [comment]
PAM 1.5.0 is out, so I'm going to try that. It needs patches from 1.5.1 as well, for auth bypass.

**Comment by ericonr** [comment]
1.5.0 is still not working, 1.5.1 will be released shortly with a security fix

**Comment by ahesford** [comment]
I was also unable to log in after updating to 1.5.1 based on this template. Socklog complains:
```
2020-11-25T20:53:36.29815 authpriv.err: Nov 25 15:53:36 login: PAM unable to dlopen(/usr/lib/security/pam_tally.so): /usr/lib/security/pam_tally.so: cannot open shared object file: No such file or directory
2020-11-25T20:53:36.29825 authpriv.err: Nov 25 15:53:36 login: PAM adding faulty module: /usr/lib/security/pam_tally.so
2020-11-25T20:53:41.43097 authpriv.notice: Nov 25 15:53:41 login: FAILED LOGIN SESSION FROM tty2 FOR ajh, Module is unknown
```
A quick look in `/etc/pam.d` reveals that `/etc/pam.d/system-login` requires the module:
```
auth       required   pam_tally.so         onerr=succeed file=/var/log/faillog
```
I see no other pam configuration files that require `pam_tally`. Commenting out this line or changing "required" to "optional" (which still prints errors) is sufficient to allow logins through `agetty`.

**Comment by ericonr** [comment]
New solution, would require a blog post + reddit PSA, ideally, once we agree that this is the course of action to take.

**Comment by ericonr** [comment]
Blocked on #26707

File: srcpkgs/pam/template [hunk 1]
```diff
@@ -22,13 +23,13 @@ conf_files="
...
-lib32disabled=yes
 short_desc="Flexible mechanism for authenticating users"
-maintainer="Orphaned <orphan@voidlinux.org>"
+maintainer="Érico Nogueira <ericonr@disroot.org>"
 license="BSD-3-Clause"
-homepage="http://www.kernel.org/pub/linux/libs/pam/"
-distfiles="http://linux-pam.org/library/Linux-PAM-${version}.tar.bz2"
-checksum=241aed1ef522f66ed672719ecf2205ec513fd0075ed80cda8e086a5b1a01d1bb
+homepage="https://github.com/linux-pam/linux-pam"
+distfiles="${homepage}/releases/download/v${version}/Linux-PAM-${version}.tar.xz"
```
Comment (ahesford) [inline srcpkgs/pam/template:30]: `changelog=https://raw.githubusercontent.com/linux-pam/linux-pam/master/NEWS`

Comment (ericonr) [reply srcpkgs/pam/template:31]: Thanks.

**Comment by ericonr** [comment]
@void-linux/pkg-committers

This is a pretty critical package, and I believe I've done everything within my power to guarantee it's working. Still, I'd appreciate testing from more than me and @ahesford.

I've been running it for a while already, so from my side it's ready to merge.

**Comment by st3r4g** [comment]
Testing this on x86_64 glibc with default pam configs, no issues so far

**Pull request merged**